mlflow/sklearn/__init__.py

```diff
@@ -884,12 +1006,77 @@ def patched_fit(original, self, *args, **kwargs):
                           for autologging (e.g., specify "fit" in order to indicate that
                           `sklearn.linear_model.LogisticRegression.fit()` is being patched)
         """
+        status = _get_autolog_training_status()
         with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                status.register_model(self, mlflow.active_run().info.run_id)
+                with status.call_scope(status.fit_scope_key):
+                    result = fit_mlflow(original, self, *args, **kwargs)
+                return result
             else:
                 return original(self, *args, **kwargs)
 
+    def patched_predict(original, self, *args, **kwargs):
+        if _SklearnTrainingSession.in_top_level():
```

Comment:
This is a safe guarding for cross validator/grid search case:

When running cross validator/grid search, the nested `model.fit` will be called in parallel, but, the `_autolog_training_status` is a global status ((including some scope flags) without thread-safe lock protecting. This safe guarding will prevent code run into this case.